addons/auth_oauth/models/auth_oauth.py

```diff
@@ -21,3 +23,12 @@ class AuthOAuthProvider(models.Model):
     css_class = fields.Char(string='CSS class', default='fa fa-fw fa-sign-in text-primary')
     body = fields.Char(required=True, help='Link text in Login Dialog', translate=True)
     sequence = fields.Integer(default=10)
+
+    @api.model
+    def _neutralize(self):
+        super()._neutralize()
+        provider_ids = self.search([('enabled', '=', True)])
+        provider_ids.update({'enabled': False})
```

Comment:
Be careful with variable names: `search()` returns *records* not *ids*
```suggestion
        providers = self.search([('enabled', '=', True)])
        providers.enabled = False
```